server/api/api/utils.py

```diff
@@ -1327,27 +1333,6 @@ async def _delete_function(
     )
 
 
-async def _update_functions_with_deletion_task_ids(
-    db_session, functions, project, background_task_name
-):
-    semaphore = asyncio.Semaphore(
-        mlrun.mlconf.background_tasks.function_deletion_batch_size
-    )
-
-    async def update_function_with_task_id(function):
-        async with semaphore:
-            await run_in_threadpool(
-                server.api.crud.Functions().set_function_deletion_task_id,
-                db_session,
-                function,
-                project,
-                background_task_name,
-            )
```

Comment:
you can keep this but use `server.api.db.session.run_function_with_new_db_session` instead.

Reply:
see `server.api.crud.runs.Runs.delete_runs` for reference

Reply:
@alonmr thanks for saying! Reworked it.